### PR TITLE
[CI] Dockerhub image push action: commit to one syntax for string manipulation.

### PR DESCRIPTION
**Pull request by NlightNFotis**
Commit to one syntax for isolating the version
number, remove the debug statements and add an
explanatory comment.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: .github/workflows/release-packages.yaml [hunk 1]
```diff
@@ -217,11 +217,10 @@ jobs:
...
       - name: Set Image Tag
         run: |
-          echo ${{ github.ref }}
-          echo $GITHUB_REF
-          echo $GITHUB_REF | cut -d "/" -f 3 >> $VERSION
-          arrVERSION=(${VERSION//-/ })
-          echo "IMAGE_TAG=diffblue/cbmc:${arrVERSION[1]}" >> $GITHUB_ENV
+          # Isolate the version number from a reference to a tag, for example,
+          # '5.20.3' from a string like 'refs/tags/cbmc-5.20.3-exp'
+          VERSION=$(echo ${{ github.ref }} | cut -d "/" -f 3 | cut -d "-" -f 2)
```
Comment (piotr-grabalski) [inline .github/workflows/release-packages.yaml:222]: is the tag reference always in that form? I wonder if that can go wrong in the future if tag is placed under different path

**Review by piotr-grabalski — APPROVED** [review]

File: .github/workflows/release-packages.yaml (hunk 1 above)
Comment (piotr-grabalski) [reply .github/workflows/release-packages.yaml:222]: Maybe use `basename`? 
something like:
`VERSION=$(basename ${{ github.ref }}  | cut -d "-" -f 2)`

Comment (NlightNFotis) [reply .github/workflows/release-packages.yaml:222]: This is related to git internals, so I doubt it's going to change anytime soon https://git-scm.com/book/en/v2/Git-Internals-Git-References

Comment (hannes-steffenhagen-diffblue) [reply .github/workflows/release-packages.yaml:222]: @piotr-grabalski As to the tagname starting with `refs/tags/`, yes, it'll be in that form. This is how it's handled in git and unlikely to change, certainly not without a major version change in git itself. As to the tagname itself, this should always be `cbmc-$VERSION`, if we're changing that of course we need to change our scripts but I'm not sure why we would change it (also we only trigger releases if the tag has that form in the first place).

**Pull request merged**